DataConnectors/JumpCloud Single Sign On/AzureFunctionJumpCloud/JCQueueTrigger1/run.ps1

```diff
@@ -0,0 +1,141 @@
+            $bytesToHash = [Text.Encoding]::UTF8.GetBytes($stringToHash)
+            $keyBytes = [Convert]::FromBase64String($sharedKey)
+            $sha256 = New-Object System.Security.Cryptography.HMACSHA256
+            $sha256.Key = $keyBytes
+            $calculatedHash = $sha256.ComputeHash($bytesToHash)
+            $encodedHash = [Convert]::ToBase64String($calculatedHash)
+            $authorization = 'SharedKey {0}:{1}' -f $customerId,$encodedHash
+            return $authorization
+        }
+        # create and post the events to Log Analytics
+            $method = "POST"
+            $contentType = "application/json"
+            $resource = "/api/logs"
+            $rfc1123date = [DateTime]::UtcNow.ToString("r")
+            $body = ([System.Text.Encoding]::UTF8.GetBytes($response))
+            $contentLength = $body.Length
+            $signature = New-BuildSignature `
+                -customerId $customerId `
+                -sharedKey $sharedKey `
+                -date $rfc1123date `
+                -contentLength $contentLength `
+                -method $method `
+                -contentType $contentType `
+                -resource $resource
+            $uri = "https://" + $customerId + ".ods.opinsights.azure.com" + $resource + "?api-version=2016-04-01"
+            $headers1 = @{
+                "Authorization" = $signature;
+                "Log-Type" = $JCTablename;
+                "x-ms-date" = $rfc1123date;
+                "time-generated-field" = $TimeStampField;
+            }
+            $result = Invoke-WebRequest -Uri $uri -Method $method -ContentType $contentType -Headers $headers1 -Body $body -UseBasicParsing
+    }
+    else{
+        Write-Output "JumpCloud: No new $JCService JumpCloud logs are avaliable as at $currentUTCtime"
+        break
+    }
+    write-output " Limit: $JCLimit; ResultCount: $JCResultCount; Totalsofar: $totalrecordcount"
+} until($JCResultCount -lt $JCLimit)
+
+#store details in function storage table to retrieve next time function runs 
+if($response.Headers.ContainsKey("X-Search_after")){
+    if($response.Headers["X-Search_after"] -ne ''){
+        #if($LastRecordTimestamp -eq ''){
+            #$LastRecordTimestamp = [System.TimeZoneInfo]::ConvertTimeBySystemTimeZoneId($(([datetime]::parseexact(($response.headers.date),"ddd, dd MMM yyyy HH:mm:ss Z",$null))), [System.TimeZoneInfo]::Local.Id, 'Greenwich Standard Time').Tostring('yyyy-MM-ddTHH:mm:ssZ')
+        #}
+        $jbody = ConvertFrom-json $response.Content
+        $LastRecordTimestamp= $jbody.timestamp[($jbody.count - 1)]
+        $LastRecordTimestamp = $LastRecordTimeStamp.ToString('yyyy-MM-ddThh:mm:ssZ')
+        $result = Add-AzTableRow -table $JCTable -PartitionKey $JCapiToken -RowKey $JCService -property @{"SearchAfter" = ("'"+$response.Headers["X-Search_after"]+"'");"StartTime"=$LastrecordTimestamp} -UpdateExisting
```

Comment:
Don't save search after in table. 

According to Jump SSO documentation:

https://jumpcloud-insights.api-docs.io/1.0/how-to-use-the-directory-insights-api/getting-all-the-pages-of-data

Please note that the X-Search_after response header value will continually change. That is, you will get a different X-Search_after value every time you specify a search. Don’t hold on to a X-Search_after value; always parse the specific response you get from the service, then add that value to the "search_after" JSON property of your query to get the next page of events.